code/modules/mob/living/simple_animal/slime/life.dm

```diff
@@ -186,12 +186,13 @@
 	if(M.stat == DEAD) // our victim died
 		if(!client)
 			if(!rabid && !attacked)
-				if(M.LAssailant && M.LAssailant != M)
+				var/mob/living/carbon/their_attacker = M.getLAssailant()
```

Comment:
Any problem if we call the weakref Resolve proc here and now without the need of this proc?

Reply:
not really it's more convenience than need